src/Tests/SDDLTests/Program.cs

```diff
@@ -230,6 +246,25 @@ private static void VerifyDescriptors()
             VerifyFileSecurityDescriptor(s_cachedTestAssetPath, "BA", "BA", 4, "A;ID;0x1200a9;;;WD", "A;ID;FA;;;SY", "A;ID;FA;;;BA", "A;ID;0x1200a9;;;BU");
         }
 
+        private static void CreateInstallStateAsset()
+        {
+            SecurityUtils.CreateSecureDirectory(s_installStateDirectory);
+            File.WriteAllLines(s_installStateFileAssetPath, new[] { "line1", "line2" });
+            SecurityUtils.SecureFile(s_installStateFileAssetPath);
+        }
+
+        private static void VerifyInstallStateDescriptors()
+        {
+            // Dump the descriptor of ProgramData since it's useful for analyzing.
+            DirectorySecurity ds = new DirectorySecurity(s_programData, s_accessControlSections);
+            string descriptor = ds.GetSecurityDescriptorSddlForm(s_accessControlSections);
+            Console.WriteLine($" Directory: {s_programData}");
+            Console.WriteLine($"Descriptor: {descriptor}");
+
+            VerifyDirectorySecurityDescriptor(s_installStateDirectory, "BA", "BA", 4, "A;OICIID;0x1200a9;;;WD", "A;OICIID;FA;;;SY", "A;OICIID;FA;;;BA", "A;OICIID;0x1200a9;;;BU");
+            VerifyFileSecurityDescriptor(s_installStateFileAssetPath, "BA", "BA", 4, "A;ID;0x1200a9;;;WD", "A;ID;FA;;;SY", "A;ID;FA;;;BA", "A;ID;0x1200a9;;;BU");
```

Comment:
I know I've seen this security descriptor notation before, but might be useful to have a comment with a link to the documentation on it. They kinda look like "magic strings" without that knowledge.

Reply:
There are links in the code (SecurityUtils and MsiPackageCache) - I just didn't duplicate the links in the manual test code. But this is the old PR. The ported on for 8.0.200 has more changes and likely the one we'll take